FlowCrypt/src/main/java/com/flowcrypt/email/jetpack/workmanager/ForwardedAttachmentsDownloaderWorker.kt

```diff
@@ -199,11 +199,17 @@ class ForwardedAttachmentsDownloaderWorker(context: Context, params: WorkerParam
                                       action: suspend (attachmentEntity: AttachmentEntity)
                                       -> InputStream?): MessageState = withContext(Dispatchers.IO) {
     var msgState = MessageState.QUEUED
-
-    val pubKeys = if (msgEntity.isEncrypted == true) {
-      SecurityUtils.getRecipientsPubKeys(applicationContext, msgEntity.allRecipients.toMutableList(),
-          account, EmailUtil.getFirstAddressString(msgEntity.from))
-    } else null
+    var pubKeys: List<String>? = null
+
+    if (msgEntity.isEncrypted == true) {
+      val senderEmail = EmailUtil.getFirstAddressString(msgEntity.from)
+      val recipients = msgEntity.allRecipients.toMutableList()
+      pubKeys = SecurityUtils.getRecipientsPubKeys(applicationContext, recipients)
+      val senderKeyDetails = SecurityUtils.getSenderKeyDetails(applicationContext,
+          account, senderEmail)
+      pubKeys.add(senderKeyDetails.publicKey
+          ?: throw NullPointerException("Sender pub key not found"))
```

Comment:
I think IllegalStateException is more appopriate

Reply:
Changed